strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/AbstractRepositoryProvider.java

```diff
@@ -158,25 +162,31 @@ protected final RepositoryOutputStream decorate(RepositoryPath repositoryPath,
             return (RepositoryOutputStream) os;
         }
 
-        String path = RepositoryFiles.stringValue(repositoryPath);
-        return RepositoryOutputStream.of(repositoryPath.getRepository(), path, os).with(this);
+        return RepositoryOutputStream.of(repositoryPath, os).with(this);
     }
 
     @Override
-    public void onBeforeWrite(RepositoryStreamContext ctx)
+    public void onBeforeWrite(RepositoryStreamContext ctx) throws IOException
     {
-        logger.debug(String.format("Writing [%s]", ctx.getPath()));
+        RepositoryPath repositoryPath = (RepositoryPath) ctx.getPath();
+        String path = RepositoryFiles.stringValue(repositoryPath);
+        
+        logger.debug(String.format("Writing [%s]", path));
+        
+        if (!RepositoryFiles.isArtifact(repositoryPath)) {
```

Comment:
@carlspring @fuss86 you should note that only Artifacts stored in DB now, as discussed within `SB-984`. But also we need make `pom.xml` to be artifacts and have `ArtifactCoordinates `within upcoming tasks.

Reply:
+1: Could you please raise a follow-up task for this?